### ISSUE TITLE
Get permissions doesn't trigger callback

### ISSUE THREAD

**Issue by thylo** [react-native-community/react-native-image-picker#581]
### Description

On Android 23+ with React Native 0.43+ (targetSDKVersion 23), the very first time you try to take a picture ou to select one from the library, the permission popups (WRITE_EXTERNAL_STORAGE and PERMISSIONS.CAMERA) shows up, but the callback is never triggered. You have to recall the method to get it working

### How to repeat issue and example

```
    gotPhotoResponse(response:{didCancel:boolean,error:{}}) {
        if (response.didCancel) {
            this.props.onCancel();
        }
        else if (response.error) {
            Alert.alert(
                I18n.t('issue_creation_take_picture_error_title'),
                I18n.t('issue_creation_take_picture_error_message'),
                [
                    {
                        text: I18n.t('issue_creation_take_picture_error_settings'),
                        onPress: this.goToSettings
                    },
                    {
                        text: I18n.t('text_cancel'),
                        onPress: this.props.onCancel
                    }
                ]
            );
        }
        else {
            this.props.onSelect(response);
        }
    }

    selectTakePhotoTapped() {
       ImagePicker.launchCamera(options, this.gotPhotoResponse.bind(this));
    }

    selectPickPhotoTapped() {
        ImagePicker.launchImageLibrary(options, this.gotPhotoResponse.bind(this));
    }

    render() {
        return <View style={styles.container}>
            <View style={styles.bigButton}>
                <Button onPress={this.selectTakePhotoTapped} title={I18n.t('issue_creation_take_picture')}/>
            </View>
            <View style={styles.smallButton}>
                <Button onPress={this.selectPickPhotoTapped}
                        title={I18n.t('issue_creation_select_picture')}/>
            </View>
        </View>
    }
```

In this case, `this.gotPhotoResponse` is never triggered after allowing or denying the permissions. Any subsequent call to `this.selectTakePhotoTapped` or `selectPickPhotoTapped` works as expected.

I tried a workaround with  checking the permissions individually with `PermissionsAndroid` from React-Native but it get messy as you can allow one permission and deny the other, launching react-native-image-picker afterwards still requires the 2 permissions in order to proceed.

**Comment by danieljoonlee** [comment]
im getting the same issue as well
```
  launchCamera = () => {
    ImagePicker.launchCamera(PhotoPicker.imagePickerOptions, this.lookupPhotoCallback);
  }

  launchImageLibrary = () => {
    ImagePicker.launchImageLibrary(PhotoPicker.imagePickerOptions, this.lookupPhotoCallback);
  }
```

the call backs dont seem to fire after turning off your permissions.  in result, the app crashes if the user tries to proceed again

**Comment by pvanliefland** [comment]
FYI, I got around the issue by "overriding" the react-native-image-picker module with a custom static class that keeps the same signature. I made a gist out of it: https://gist.github.com/pvanliefland/92046b17444a8760475e7cc0dae4fec7

If the library author is interested, I can work on a pull request.

**Comment by jwrubel** [comment]
I started testing against a new Nexus 6 build with API 26 and ran across this issue - on first load, the app never gets to the response on the first run even if you accept all the permissions warnings, so my app just hangs. @pvanliefland your gist worked perfectly - thanks mate! I'm really shocked that no one else using this library has run across the same problem.

**Comment by leograde** [comment]
I also can't believe that so few people have actually run across this issue. It would be nice if this permission functionality made by @pvanliefland came out of the box. 

@pvanliefland Thank you so much for posting this. It helped a ton!

**Comment by arelstone** [comment]
@pvanliefland very much love from me, I was facing the same issue.
Great job. Although I cannot get your gist to work with IOS. For now I'll just do some platform specific.

**Comment by Alexisvt** [comment]
@arelstone can you please share with us your solution for iOS, currently Im working on project where we need this component for both platform android and iOS, thanks!

**Comment by jwrubel** [comment]
For what it is worth, @pvanliefland 's gist worked fine for me on both platforms without any changes needed.

**Comment by arelstone** [comment]
@Alexisvt I imported both react-native-image-picker and the gist provided by @pvanliefland and used Platform.OS to handle which one to use

```js
import RNImagePicker from 'react-native-image-picker';
import ImagePickerAndroidWrapper from 'ImagePickerFromGist';

// Android doesn't get a callback when asking for permission.
// @see https://github.com/react-community/react-native-image-picker/issues/581
// `ImagePickerFromGist` is a wrapper for handling this.
const ImagePicker = Platform.OS === 'ios' ? RNImagePicker : ImagePickerAndroidWrapper;

export function launchImageLibrary(options) {
	return new Promise((resolve, reject) => {
		ImagePicker.launchImageLibrary(options, response => {
			if (response.didCancel) {
				return resolve();
			}
			if (response.error) {
				return reject(response.error);
			}
			resolve(response);
		});
	});
}
```

**Comment by Alexisvt** [comment]
Thanks @arelstone !!

**Comment by mbret** [comment]
I guess we all have that issue... Thank for you gist.

**Comment by klcantrell** [comment]
I'm on React Native 0.57.1 with Android 26 (targetSdkVersion = 26) and I'm not experiencing this issue.  @thylo did this perhaps get solved in #808 and should this issue then be closed?
